blog/_posts/2025-07-15-network-perf-aks.md

```diff
@@ -0,0 +1,89 @@
+---
+title: "Performance Tuning AKS for Network Intensive Workloads "
+description: "Learn more about how to optimize network performance on AKS nodes through benchmark and comparison."
+date: 2025-07-15 # date is important. future dates will not be published
+authors:
+- Anson Qian
+- Alyssa Vu
+categories:
+- performance
+- networking
+---
+
+## Background
+
+As more intelligent applications are deployed and hosted on Azure Kubernetes Service (AKS), network performance becomes increasingly critical to ensuring a seamless user experience. For example, a chatbot server running in an AKS cluster needs to handle high volumes of network traffic with low latency, while retrieving contextual data — such as conversation history and user feedback from a database or cache, and interacting with a LLM (Large Language Model) endpoint through prompt requests and streamed inference responses.
+
+In this blog post, we share how we conducted simple benchmarks to evaluate and compare network performance across various VM (Virtual Machine) SKU and series. We also provide recommendations on key kernel settings to help you explore the trade-offs between network performance and resource usage.
+
+## Benchmark
+
+Our methodology involves conducting tests and measurements to identify key factors affecting network performance for applications running on AKS. We simulated a common use case: a pair of pods communicating in TCP protocol across two different nodes within the same AKS cluster. We measured various performance metrics, including throughput, round-trip time (RTT), and retransmission rate in the presence of packet loss at high bandwidth usage.
```

Comment:
nit - but we use AKS as the test bed even though we also used EKS.  Could say `k8s` or `cluster` if we want to genericize it, but only nit.

Reply:
Done